apps/confidential/confidential.go

```diff
@@ -592,6 +600,46 @@ func (cca Client) AcquireTokenSilent(ctx context.Context, scopes []string, opts
 	return cca.base.AcquireTokenSilent(ctx, silentParameters)
 }
 
+// acquireTokenByUsernamePasswordOptions contains optional configuration for AcquireTokenByUsernamePassword
+type acquireTokenByUsernamePasswordOptions struct {
+	claims, tenantID string
+	authnScheme      AuthenticationScheme
+}
+
+// AcquireByUsernamePasswordOption is implemented by options for AcquireTokenByUsernamePassword
+type AcquireByUsernamePasswordOption interface {
```

Comment:
We would need some unit tests where the HTTP is mocked. The tests should cover use of `AcquireTokenByUsernamePassword` and `AcquireTokenSilent`

Reply:
I haven't changed AcquireTokenSilent. I added tests for AcquireTokenByUsernamePassword